go/vt/vtctl/reparentutil/planned_reparenter.go

```diff
@@ -490,22 +517,36 @@ func (pr *PlannedReparenter) reparentShardLocked(
 
 	currentPrimary := FindCurrentPrimary(tabletMap, pr.logger)
 	reparentJournalPos := ""
+	// needsReload is used to keep track whether we need to refresh the state
+	// of the new primary tablet. The only case that we need to reload the state
+	// is when we are initializing the new primary. The reason is that the first
+	// time we try to setup all the components like vreplication.Engine, they fail
+	// since the database isn't created until we setServing.
+	// A call to Refresh state fixes all the components. This isn't strictly necessary
+	// in the sense that all the components will retry initialization anyways after some
+	// time, so even without a call to RefreshState, they all converge correctly.
+	needsReload := false
```

Comment:
Let's rename this to `needsRefresh`. reload makes me think we are reloading topo information.

Reply:
fixed